qualtran/bloqs/data_loading/qrom_abc.py

```diff
@@ -0,0 +1,294 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Base class for Bloqs implementing a QROM (Quantum read-only memory) circuit."""
+import abc
+from functools import cached_property
+from typing import Dict, Tuple, Type, TypeVar, Union
+
+import attrs
+import cirq
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from qualtran import BoundedQUInt, QAny, Register
+from qualtran.simulation.classical_sim import ClassicalValT
+from qualtran.symbolics import bit_length, is_symbolic, shape, Shaped, SymbolicInt
+
+QROMT = TypeVar('QROMT', bound='QROMABC')
+
+
+@cirq.value_equality(distinct_child_types=True)
+@attrs.frozen
+class QROMABC(abc.ABC):
```

Comment:
QROMInterface or something else, I'm not mad on ABC

Reply:
BaseQROM or _BaseQROM or QROMBase